compiler/codegen/OMRCodeGenerator.hpp

```diff
@@ -489,8 +489,38 @@ class OMR_EXTENSIBLE CodeGenerator
    bool isRegisterClobberable(TR::Register *reg, uint16_t count);
 
    // ilgen
-   bool ilOpCodeIsSupported(TR::ILOpCodes); // no virt, default, cast
 
+   /**
+    * @brief Returns if an IL OpCode is supported by current CodeGen
+    *
+    * @param op The IL OpCode being checked.
+    *
+    * @return True if the IL OpCode is supported otherwise false.
+    */
+   static bool ilOpCodeIsSupported(TR::ILOpCodes op);
+
+   /**
+    * @brief Returns the corresponding IL OpCode for an intrinsic method
+    *
+    * This query maps an intrinsic method to an IL OpCode, with the requirement that
+    * the method's child(ren) corresponds to the OpCode's child(ren) exactly.
+    * It is usually used by the IL Gen transforming the intrinsic method to IL OpCode
+    * so that it can leverage existing framework for better optimization.
+    *
+    * @param method The intrinsic method being checked.
+    *
+    * @return The corresponding IL OpCode for the intrinsic method.
+    */
+   static TR::ILOpCodes ilOpCodeForIntrinsicMethod(TR::RecognizedMethod method) { return TR::BadILOp; }
+
+   /**
+    * @brief Returns if an intrinsic method is supported by current CodeGen
+    *
+    * @param method The intrinsic method being checked.
+    *
+    * @return True if the intrinsic method is supported otherwise false.
+    */
+   static inline bool intrinsicMethodSupported(TR::RecognizedMethod method);
```

Comment:
This function is named inconsistently too.  Since it answers a question, how about `isIntrinsicMethodSupported` ?

Reply:
Good suggestion, I've updated its name to `isIntrinsicMethodSupported`.